lib/query_relation/array_based_collection.rb

```diff
@@ -0,0 +1,14 @@
+class QueryRelation
+  module ArrayBasedCollection
+    # Default filter for an array based +collection+ based on various hash
+    # based +options+ that are used by QueryRelation.
+    #
+    def filter_array_collection(collection, options)
+      collection = collection.drop(options[:offset]) if options[:offset]
+      collection = collection.take(options[:limit]) if options[:limit]
+      collection = collection.select{ |hash| hash.slice(*options[:where].keys) == options[:where] } if options[:where]
+      collection = collection.sort_by{ |hash| hash[options[:order]] } if options[:order]
```

Comment:
can we assume there is only 1 column in order? (think we can but checking)

Reply:
I say for a default implementation we can. If users need something fancier they can implement it.